src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1895,9 +1887,8 @@ public static SqlConnection OpenSqlConnection(ConnectionInfo connInfo, string fe
                 // allow pooling connections for language service feature to improve intellisense connection retention and performance.
                 bool shouldForceDisablePooling = !EnableConnectionPooling && featureName != Constants.LanguageServiceFeature;
 
-                // increase the connection and command timeout to at least 30 seconds and and build connection string
-                connInfo.ConnectionDetails.ConnectTimeout = Math.Max(30, connInfo.ConnectionDetails.ConnectTimeout ?? 0);
-                connInfo.ConnectionDetails.CommandTimeout = Math.Max(30, connInfo.ConnectionDetails.CommandTimeout ?? 0);
+                // increase the connection and command timeout to at least 30 seconds and set application name.
+                FillInDefaultDetailsForConnections(connInfo.ConnectionDetails, featureName);

```

Comment:
Reassign connInfo.ConnectionDetails here - by returning updated connection details from the method.